File: tests/typesafety/typescript/newtype_arithmetic.ts
// expect: passes

declare const secondsBrand: unique symbol;
declare const kilogramsBrand: unique symbol;

type Seconds = number & { readonly [secondsBrand]: true };
type Kilograms = number & { readonly [kilogramsBrand]: true };

export function nonsense(duration: Seconds, load: Kilograms): number {
  return duration + load; // accepted, and the sum means nothing
}

File: tests/typesafety/python/newtype_arithmetic.py
# expect: passes
# title: Seconds plus Kilograms still compiles, and the sum is a meaningless int
from typing import NewType

Seconds = NewType("Seconds", int)
Kilograms = NewType("Kilograms", int)


def nonsense(duration: Seconds, load: Kilograms) -> int:
    return duration + load  # accepted, and the sum means nothing
